### PR TITLE
Fixed duplicate chpl nbr err msg being displayed when editing product

### PR DESCRIPTION
**Pull request by tmy1313**
OCD-2343

### CODE REVIEW

**Comment by kekey1** [comment]
@andlar I think there actually IS a unit test that was failing that would (may?) have caught this. gov.healthit.chpl.questionableActivity.ProductTest.testEditOwnerHistoryItem has been failing (was in chpl-service, now chpl-api). I'd like to know if it's passing now with this change in Todd's environment. @tmy1313 can you update your jenkins jobs so we can see about the unit test? You can look at mine or the chpladmin ones - basically we don't need separate chpl-service and chpl-auth jenkins jobs but rather one top-level chpl build that will do the necessary mvn installs and run all the tests in all the sub-modules.

File: PENDING_RELEASE_NOTES.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+## Version 15.15.2
+_Date TBD_
+
+### Bugs Fixed
+* Fixed bug where user erroneously receives a Duplicate CHPL PRoduct Number error when editting a product
```
Comment (andlar) [inline PENDING_RELEASE_NOTES.md]: P`R`oduct -> P`r`oduct

File: chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/ProductController.java [hunk 2]
```diff
@@ -324,8 +324,10 @@
 
             for (CertifiedProductDetailsDTO filterCp : filteredResults) {
                 //Add it to the list of duplicates
-                duplicateChplProductNumbers.add(new DuplicateChplProdNumber(
-                    cpDTO.getChplProductNumber(), filterCp.getChplProductNumber(), filterCp.getChplProductNumber()));
+                if (!cpDTO.getChplProductNumber().equals(filterCp.getChplProductNumber())) {
```
Comment (andlar) [inline chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/ProductController.java]: This is a string comparison, right? Don't we usually do string comparisons with some sort of StringUtil operator in some way? @kekey1 am I crazy?

File: chpl/chpl-api/src/test/java/gov/healthit/chpl/web/controller/ProductControllerTest.java [hunk 3]
```diff
@@ -111,4 +112,31 @@ public void testUpdateDuplicateChplProductNumberValidationException() throws Jso
         
         productController.updateProduct(request);
     }
+	
+	@Transactional
+	@Test
+	@Rollback(true)
+	public void testUpdateProductContactInformation() throws EntityRetrievalException, JsonProcessingException, EntityCreationException, InvalidArgumentsException, ValidationException {
```
Comment (andlar) [inline chpl/chpl-api/src/test/java/gov/healthit/chpl/web/controller/ProductControllerTest.java]: Does this test have anything to do with the feature in this PR? Or was it just added for some other reason?

Comment (tmy1313) [reply chpl/chpl-api/src/test/java/gov/healthit/chpl/web/controller/ProductControllerTest.java]: There was no test in this class to test when the product's contact information was being updated, so I added it.  This was the scenario called out in the ticket - updating a product's contact info caused an error.  I wrote the test before making the change, it failed.  Made the change and it passed.  I'm not sure what I did wrong here...

File: chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/ProductController.java (hunk 2 above)
Comment (kekey1) [reply chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/ProductController.java]: I think you might be thinking of the StringUtils.isEmpty methods that we use (avoids checking for both null and blank strings - and forgetting one or the other of those possibilities - all over the place). I'm pretty sure that for string comparisons we just use equals or equalsIgnoreCase as applicable.

File: chpl/chpl-api/src/test/java/gov/healthit/chpl/web/controller/ProductControllerTest.java (hunk 3 above)
Comment (andlar) [reply chpl/chpl-api/src/test/java/gov/healthit/chpl/web/controller/ProductControllerTest.java]: I didn't say you did anything wrong. It's just that the code that changed doesn't seem to have any relationship to contact information changing, and that looked odd. If the test failed initially, then the code was changed, and the test passed, then that's TDD (or at least the first two steps). Unless you see anything that should need refactoring then it's good

**Pull request merged**